modules/api/php/endpoints/candidate/visit/instruments.class.inc

```diff
@@ -166,6 +172,59 @@ class Instruments extends Endpoint implements \LORIS\Middleware\ETagCalculator
         return $this->_cache;
     }
 
+    /**
+     * Handle an incoming post request.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePOST(ServerRequestInterface $request): ResponseInterface
+    {
+        $version = $request->getAttribute("LORIS-API-Version");
+        $user    = $request->getAttribute("user");
+        $loris   = $request->getAttribute("loris");
+
+        if ($version == 'v0.0.3') {
+            return new \LORIS\Http\Response\JSON\NotFound();
+        }
+
+        $requestdata = json_decode($request->getBody()->__toString(), true);
+        if (empty($requestdata)) {
+            return new \LORIS\Http\Response\JSON\BadRequest();
+        }
+        foreach ($requestdata['Instruments'] as $instrument) {
+            try {
+                $instrument = \NDB_BVL_Instrument::factory(
+                    $loris,
+                    $instrument,
+                    '',
+                    '',
+                    true
+                );
+                if (!$instrument->_hasAccess($user)) {
+                    return new \LORIS\Http\Response\JSON\Forbidden();
+                }
+            } catch (\Exception $e) {
+                return new \LORIS\Http\Response\JSON\InternalServerError();
```

Comment:
Probably a good place to use `422  Unprocessable Entity`
A plain 500 without message is dry and not helpful

Reply:
But an exception would be an error on the server side, not the client side.